kube/charts/helm/helm.go

```diff
@@ -28,44 +28,38 @@ import (
 	loader "helm.sh/helm/v3/pkg/chart/loader"
 	env "helm.sh/helm/v3/pkg/cli"
 	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
 // Actions helm actions
 type Actions struct {
-	Config       *action.Configuration
-	Settings     *env.EnvSettings
-	kubeConnInit bool
+	Config    *action.Configuration
+	Namespace string
 }
 
 // NewHelmActions new helm action
-func NewHelmActions(settings *env.EnvSettings) *Actions {
-	if settings == nil {
-		settings = env.New()
+func NewHelmActions(getter genericclioptions.RESTClientGetter) (*Actions, error) {
+	if getter == nil {
+		settings := env.New()
+		getter = settings.RESTClientGetter()
 	}
-	return &Actions{
-		Config:       new(action.Configuration),
-		Settings:     settings,
-		kubeConnInit: false,
-	}
-}
 
-func (hc *Actions) initKubeClient() error {
-	if hc.kubeConnInit {
-		return nil
+	namespace := "default"
+	if ns, _, err := getter.ToRawKubeConfigLoader().Namespace(); err == nil {
+		namespace = ns
 	}
-	if err := hc.Config.Init(
-		hc.Settings.RESTClientGetter(),
-		hc.Settings.Namespace(),
-		"configmap",
-		log.Printf,
-	); err != nil {
-		log.Fatal(err)
+	actions := &Actions{
+		Config: &action.Configuration{
+			RESTClientGetter: getter,
+		},
+		Namespace: namespace,
```

Comment:
Do we need to set Namespace here, in addition to pass Namespace in Init() the next line ? (maybe some constraint form the helm API ? )

Reply:
We need it for the install action too.